main/main.cpp

```diff
@@ -44,9 +50,44 @@ static void initResources()
 
 int main(int argc, char** argv)
       {
+      // Force the 8-bit text encoding to UTF-8. This is the default encoding on all supported platforms except for MSVC under Windows, which
+      // would otherwise default to the local ANSI code page and cause corruption of any non-ANSI Unicode characters in command-line arguments.
+      QTextCodec::setCodecForLocale(QTextCodec::codecForName("UTF-8"));
+
       initResources();
 
       ModulesSetup::instance()->setup();
 
-      return Ms::runApplication(argc, argv);
+#if (defined (_MSCVER) || defined (_MSC_VER))
+      // On MSVC under Windows, we need to manually retrieve the command-line arguments and convert them from UTF-16 to UTF-8.
+      // This prevents data loss if there are any characters that wouldn't fit in the local ANSI code page.
+      int argcUTF16 { 0 };
+      LPWSTR* argvUTF16 { CommandLineToArgvW(GetCommandLineW(), &argcUTF16) };
+
+      std::vector<QByteArray> argvUTF8Q;
+
+      std::for_each(argvUTF16, argvUTF16 + argcUTF16, [&argvUTF8Q](const auto& arg) {
+            argvUTF8Q.emplace_back(QString::fromUtf16(reinterpret_cast<const char16_t*>(arg), -1).toUtf8());
+            });
+
+      LocalFree(argvUTF16);
+
+      // Ms::runApplication() wants an argv-style array of raw pointers to the arguments, so let's create a vector of them.
+      std::vector<char*> argvUTF8;
+
+      for (auto& arg : argvUTF8Q)
+            argvUTF8.push_back(arg.data());
+
+      // Don't use the arguments passed to main(), because they're in the local ANSI code page.
+      Q_UNUSED(argc);
+      Q_UNUSED(argv);
+
+      int argcFinal { argcUTF16 };
+      char** argvFinal { argvUTF8.data() };
+#else
+      int argcFinal { argc };
```

Comment:
It would be good to test whether this may lead to crashes at application exit. `Ms::runApplication` takes a *reference* to `argc` rather than a value since it has to pass it to [`QApplication`](https://doc.qt.io/qt-5/qapplication.html#QApplication) constructor. `QApplication` requires that those `argc` and `argv` remain valid for the entire `QApplication` lifecycle so I wonder whether declaring them in `main()` function scope is sufficient for that. Should likely be OK but still better to test.

Reply:
No crash on exit with MSVC

Reply:
I just spent time time checking this, and :

`QCoreApplication::exit()` is called inside `QCoreApplication::quit()`, which is called inside `MuseScore::closeEvent()`, which is called several levels inside `QCoreApplication::exec()`, which is called inside `Ms::runApplication()`, which is called in `main()`, where all of `argvUTF8Q`, `argvUTF8`, `argcFinal`, and `argvFinal` are in scope and alive.

Further, the documentation for `QCoreApplication::exec()` states:

“We recommend that you connect clean-up code to the `QCoreApplication::aboutToQuit()` signal, instead of putting it in your application's `main()` function because on some platforms the `exec()` call may not return. For example, on Windows when the user logs off, the system terminates the process after Qt closes all top-level windows. Hence, there is no guarantee that the application will have time to exit its event loop and execute code at the end of the `main()` function after the `exec()` call.”

In other words, I'm pretty confident that this is safe.

Reply:
Please still fix the commit title